src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Enums.cs

```diff
@@ -167,23 +168,122 @@ public enum FloatComparisonMode : byte
         TrueUnorderedSignaling = 31,
     }
 
+    [Flags]
     public enum StringComparisonMode : byte {
+        /// <summary>
+        /// _SIDD_CMP_EQUAL_ANY
+        /// </summary>
+        EqualAny = 0x00,
+
+        /// <summary>
+        /// _SIDD_CMP_RANGES
+        /// </summary>
+        Ranges = 0x04,
+
+        /// <summary>
+        /// _SIDD_CMP_EQUAL_EACH
+        /// </summary>
+        EqualEach = 0x08,
+
+        /// <summary>
+        /// _SIDD_CMP_EQUAL_ORDERED
+        /// </summary>
+        EqualOrdered = 0x0c,
+
+        /// <summary>
+        /// _SIDD_NEGATIVE_POLARITY
+        /// </summary>
+        NegativeResult = 0x10,
+
+        /// <summary>
+        /// _SIDD_MASKED_NEGATIVE_POLARITY
+        /// </summary>
+        NegativeUsefulResult = 0x30,
```

Comment:
The architecture manual refers to this as `Masked (-)`. We should likely also explicitly expose an equivalent for `Masked (+)`

Reply:
Here `useful` already means "masked" or "selected".